python_modules/libraries/dagster-dg/dagster_dg/component.py

```diff
@@ -197,7 +206,28 @@ def _fetch_plugin_manifest(context: "DgContext", args: list[str]) -> PluginManif
         )
     else:
         result = context.external_components_command(["list", "plugins", *args])
-        return deserialize_value(result, as_type=PluginManifest)
+        result = deserialize_value(result, as_type=Union[SerializableErrorInfo, PluginManifest])
```

Comment:
wondering if we need to be more defensive here if either:
- some stray output snuck into stdout (akin to the dbt thing from earlier this week)
- we're on an older version that handled errors differently? I guess this wouldn't make that case any worse, it would be some kind of deserialization error right? but we could have some fallback logic where we just pipe out the output if it's unexpected for some reason, and/or filter out lines that aren't valid serdes

Reply:
>some stray output snuck into stdout (akin to the dbt thing from earlier this week)

yup we need to make list plugins use the same mechanism as list defs

>we're on an older version that handled errors differently? 

- If we're on an older version of `dagster` then I believe the old behavior should still occur, because I didn't change the code that handles subprocess failure (which is what should happen)
- Using an older version of `dg` with newer `dagster` is now disallowed